schema.sql

```diff
@@ -1,76 +1,88 @@
+	PRIMARY KEY(id)
 );
 
--- add species to animals table
-ALTER TABLE
-    animals
-ADD
-    COLUMN species VARCHAR(250);
+/* adding column species to the animals table*/
+
+ALTER TABLE animals 
+ADD COLUMN species VARCHAR(100);
 
--- add new table owners
 CREATE TABLE owners (
-    id INT GENERATED ALWAYS AS IDENTITY,
-    full_name VARCHAR(250),
-    age INT,
-    PRIMARY KEY(id)
+	id int GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+	full_name varchar(100),
+	age int
 );
 
--- add new table species 
 CREATE TABLE species (
-    id INT GENERATED ALWAYS AS IDENTITY,
-    name VARCHAR(250),
-    PRIMARY KEY(id)
+	id int GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+	name varchar(100)
+);
+
+ALTER TABLE animals DROP COLUMN species;
+
+ALTER TABLE animals 
+ADD COLUMN species_id int;
+
+ALTER TABLE animals
+ADD FOREIGN KEY (species_id) 
+REFERENCES species(id)
+ON DELETE CASCADE;
+
+
+ALTER TABLE animals 
+ADD COLUMN owners_id int;
+
+ALTER TABLE animals
+ADD FOREIGN KEY (owners_id) 
+REFERENCES owners(id)
+ON DELETE CASCADE;
+
+----------
+
+CREATE table vets (
+	id int GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
+	name varchar,
+	age int,
+	date_of_graduation date
 );
 
--- update animals table 
-ALTER TABLE
-    animals DROP COLUMN species;
-
-ALTER TABLE
-    animals
-ADD
-    COLUMN species_id INT,
-ADD
-    CONSTRAINT species_key FOREIGN KEY (species_id) REFERENCES species(id);
-
-ALTER TABLE
-    animals
-ADD
-    COLUMN owner_id INT,
-ADD
-    CONSTRAINT owner_key FOREIGN KEY (owner_id) REFERENCES owners(id);
-
--- add vets table 
-CREATE TABLE vets (
-    id INT GENERATED ALWAYS AS IDENTITY,
-    name VARCHAR(250),
-    age INT,
-    date_of_graduation DATE,
-    PRIMARY KEY(id)
+
+CREATE table specializations (
+	species_id int,
+	vets_id int,
+	PRIMARY KEY (species_id, vets_id),
+	FOREIGN KEY (species_id) REFERENCES species(id),
+	FOREIGN KEY (vets_id) REFERENCES vets(id)
 );
 
--- add specializations table
-CREATE TABLE specializations (
-    vet_id INT,
-    species_id INT,
-    FOREIGN KEY (vet_id) REFERENCES vets(id),
-    FOREIGN KEY (species_id) REFERENCES species (id),
-    PRIMARY KEY (vet_id, species_id)
+create table visits (
+	animals_id int,
+	vets_id int,
+  visit_date date,
+	PRIMARY KEY (animals_id, vets_id, visit_date),
+	FOREIGN KEY (animals_id) REFERENCES animals(id),
+	FOREIGN KEY (vets_id) REFERENCES vets(id)
 );
 
--- add visits table
-CREATE TABLE visits (
-    animal_id INT,
-    vet_id INT,
-    visit_date DATE,
-    FOREIGN KEY (animal_id) REFERENCES animals (id),
-    FOREIGN KEY (vet_id) REFERENCES vets (id)
-);
\ No newline at end of file
+
+ALTER TABLE visits 
+DROP CONSTRAINT visits_pkey
+
+-- Add an email column to your owners table
+ALTER TABLE owners ADD COLUMN email VARCHAR(120);
+
+-- Performance Audit Indexing
+--for Visits Table
+CREATE INDEX visits_animal_id ON visits(animals_id);
+CREATE INDEX visits_vets_id ON visits(vets_id);
+
+--for Owners Table
+CREATE INDEX owner_email ON owners(email);
```

Comment:
## Required Change :recycle:

![image](https://user-images.githubusercontent.com/93039359/222041931-d9081966-baa5-42ce-86e3-4a2a3d8f7d6e.png)

- [ ] It is really great that you have added Indexes which have made your queries efficient :+1: but as the previous reviewer suggested and also it is a requirement of your project Before adding indexes your query execution time should be more than 1000ms then you would find a solution to make it lower than 1000 and after finding the solution you will add the solution to your schema.sql file :+1: You have done everything well but you missed the first point that your queries should have execution time more than 1000ms and you can do it by populating your database with more data :+1:

## Additional Information :memo:

**:red_circle: Question**

I have already made my queries efficient which was the goal of this whole project then why are you asking me to increase my before-execution time?

**:green_circle: Answer**
1. This is a project requirement means it has something really important for you to learn :thinking:
2. The important for you to learn is that when you will be working as a Full Stack Engineer you will be managing databases too and those databases would be very big :fearful: in that case, you can never think how much execution time you would have that time means it would be unimaginable :fearful: so you would be given tasks to reduce that execution time to a reasonable execution time let's say 1000ms because it sounds enough. So, to make you ready for that situation microverse wants to make you ready because most of the time you will be managing the databases in your Full Stack Development Job :+1:

## Contact Me :mobile_phone_off:

Do let me know if you have any questions :+1: If you wanna discuss with me then search my name **Hammas** in the **full-time-students** channel :blush: